components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileDefaultMoveExistingFileStrategy.java

```diff
@@ -49,14 +49,24 @@ public boolean moveExistingFile(GenericFileEndpoint endpoint, GenericFileOperati
         dummy.getIn().setHeader(Exchange.FILE_PARENT, parent);
 
         String to = endpoint.getMoveExisting().evaluate(dummy, String.class);
-        // we must normalize it (to avoid having both \ and / in the name which
-        // confuses java.io.File)
-        to = FileUtil.normalizePath(to);
+
         if (ObjectHelper.isEmpty(to)) {
             throw new GenericFileOperationFailedException(
                     "moveExisting evaluated as empty String, cannot move existing file: " + fileName);
         }
 
+        if (to.length() > 1 && to.endsWith("/")) {
```

Comment:
I see some of this logic to check the name being repeated across the three strategies, can we somehow unify this logic into a common function to ease maintainability?

Reply:
the problem is where to put the common logic, FileUtils doesn't seem a good fit.

Reply:
By creating a singleton utility class in the component perhaps?